Content.Server/Bed/BedSystem.cs

```diff
@@ -119,6 +129,14 @@ private void OnEmagged(EntityUid uid, StasisBedComponent component, ref GotEmagg
             args.Handled = true;
         }
 
+        private void OnIsRotting(EntityUid uid, BuckleComponent component, ref IsRottingEvent args)
+        {
+            if (args.Handled)
+                return;
+            args.Handled = component is { Buckled: true, BuckledTo: not null } &&
+                           HasComp<ActiveAntiRottingOnBuckleComponent>(component.BuckledTo.Value);
```

Comment:
instead of only checking if the component is present on the bed entity, you should instead have a RequiresPower boolean on the `ActiveAntiRottingOnBuckleComponent`  and check if it's powered here. (or cache it by subbing to a OnPowerChangedEvent.

This way, the component can simply be added to any buckleable entity and doesn't need to be tied to stasis beds.

Reply:
@EmoGarbage404, yep, that makes sense. I've updated the PR.